src/components/DataElement/Recent.jsx

```diff
@@ -10,14 +13,46 @@ import {
 import {Label3, Paragraph3, Paragraph4} from 'baseui/typography';
 import {StyledLink} from 'baseui/link';
 import {ProgressBar} from 'baseui/progress-bar';
-import {useStyletron} from 'baseui';
 
-export default function Recent({isLoading, data}) {
+function SearchIcon() {
+  const [css, theme] = useStyletron();
+  return (
+    <div
+      className={css({
+        display: 'flex',
+        alignItems: 'center',
+        paddingLeft: theme.sizing.scale500,
+      })}
+    >
+      <Search size="18px" />
+    </div>
+  );
+}
+
+const compare = (text = '', filterText = '') =>
+  text.toLowerCase().includes(filterText.toLowerCase());
+
+export default function Recent({ isLoading, data }) {
   const [, theme] = useStyletron();
+  const [filter, setFilter] = useState('');
+  const filteredData = data
+    ? data.filter(
+        ({ city, date }) => compare(city, filter) || compare(date, filter)
+      )
+    : [];
 
   return (
     <>
-      <Label3>Ostatnie</Label3>
+      <Label3 $style={{ marginBottom: '12px' }}>
+        Ostatnie
+      </Label3>
+      <Input
+        size={SIZE.compact}
+        overrides={{Before: SearchIcon}}
```

Comment:
Base Web's `Input` component has a `startEnchancer` prop that you can use to get the same result. You can use it like this: `startEnhancer={() => <Search />}`. You could then remove redundant `SearchIcon` component.

Reply:
You're right, but using startEnhancer prop it will look slightly different. In my opinion the current variant looks a bit better, but I can change it if you want (more docs here: https://baseweb.design/components/input/#before-and-after-overrides)

Reply:
You're right! I thought this is some "hacky" way to get the same effect. 😆